src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/types.ts

```diff
@@ -41,17 +41,22 @@ interface PublishesOptions {
   totalCardinality$: PublishingSubject<number>;
 }
 
-export type OptionsListComponentApi = OptionsListControlApi &
-  PublishesOptions &
+export type OptionsListStateApis = StateManager<SelectionsState>['api'] &
   StateManager<DefaultDataControlState>['api'] &
   StateManager<EditorState>['api'] &
-  StateManager<SelectionsState>['api'] &
   StateManager<TemporaryState>['api'] & {
+    sort$: PublishingSubject<OptionsListSortingType | undefined>;
```

Comment:
nit: why specify sort & setSort here instead of having `sort` exist in the `TemporaryState` type?

Reply:
Sort is apparently not handled by the temporary state manager, and in fact doesn't use a state manager at all. Looks like it's because we want to export it into `rawState` when serializing.